library/src/main/java/com/yayandroid/locationmanager/providers/locationprovider/DispatcherLocationProvider.java

```diff
@@ -212,4 +212,35 @@ private DispatcherLocationSource getSourceProvider() {
         }
         return dispatcherLocationSource;
     }
+    
+     @Override
+    public void onConnected(Bundle bundle) {
+
+    }
+
+    @Override
+    public void onConnectionSuspended(int i) {
+        if(activeProvider instanceof GooglePlayServicesLocationProvider){
+            activeProvider.cancel();
```

Comment:
And also you don't need to do this stuff manually, it will be done via [scheduled task](https://github.com/yayaa/LocationManager/blob/master/library/src/main/java/com/yayandroid/locationmanager/providers/locationprovider/DispatcherLocationProvider.java#L87) when waiting time is up. 

Of course, this would mean quick fallback, agree on that one. But in that case, this is not a concern of GooglePlayServices so using SourceListener which belongs to GooglePlayServicesProvider is not a right tool.

You could create another interface just to notify back when it would fail. And i would do that within another PR, because those 2 different topics now you're trying to cover.